sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgInstructionFactory.cs

```diff
@@ -180,6 +182,23 @@ private IEnumerable<Instruction> ProcessObjectCreationExpression(ObjectCreationE
                     new[] { expressionService.GetExpression(objectCreationExpression) }.Concat(arguments).ToArray()));
         }
 
+        private IEnumerable<Instruction> ProcessArrayCreationExpression(ArrayCreationExpressionSyntax arrayCreationExpression)
+        {
+            var arrayTypeSymbol = semanticModel.GetTypeInfo(arrayCreationExpression).Type as IArrayTypeSymbol;
+            if (arrayTypeSymbol == null)
+            {
+                return NoInstructions;
```

Comment:
You should associate the `arrayCreationExpression` with `const` here. Otherwise the following could throw:
```
Foo(new string[]);
```

Reply:
I've added a test that creates an array of a type that doesn't exist, thinking it might hit this code path. It doesn't - we can still get the _ArrayTypeSymbol_, we just don't know what the _ElementType_ is.